app/javascript/packages/transliterable-field-group/index.ts

```diff
@@ -0,0 +1,375 @@
+import { trackError } from '@18f/identity-analytics';
+import { request } from '@18f/identity-request';
+
+/**
+ * Representation of the response from a transliteration API call
+ */
+interface ValidationResponse {
+  success: boolean;
+  data: Record<string, string>;
+}
+
+/**
+ * Internal cache structure for controlling when API call is made
+ * and reusing result
+ */
+interface ValidationResultCache {
+  preValidationValues: Record<string, string>;
+  postValidationValues: Record<string, string>;
+  result: Record<string, string>;
+}
+
+/**
+ * This adds an asynchronous transliteration check for certain form fields
+ * that gets run between form validation and submission.
```

Comment:
is form validation occurring before the user hits submit?

Reply:
Yes, validation may occur at two points. First is 1.5 seconds after the last change to a transliterable input. Second is when the `submit` button is clicked, if the prior 1.5 seconds did not pass yet. In both cases, validation with the backend will occur before both browser validation and submission.